data-prepper-plugins/s3-source/src/integrationTest/java/com/amazon/dataprepper/plugins/source/SqsWorkerIT.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.plugins.source.configuration.OnErrorOption;
+import com.amazon.dataprepper.plugins.source.configuration.SqsOptions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.sqs.SqsClient;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class SqsWorkerIT {
+    private SqsClient sqsClient;
+    private S3Service s3Service;
+    private S3SourceConfig s3SourceConfig;
+    private S3ObjectGenerator s3ObjectGenerator;
+    private String bucket;
+
+    @BeforeEach
+    void setUp() {
+        final S3Client s3Client = S3Client.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+        bucket = System.getProperty("tests.s3source.bucket");
+        s3ObjectGenerator = new S3ObjectGenerator(s3Client, bucket);
+
+        sqsClient = SqsClient.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+
+        s3SourceConfig = mock(S3SourceConfig.class);
+        s3Service = mock(S3Service.class);
+
+        final SqsOptions sqsOptions = mock(SqsOptions.class);
+        when(sqsOptions.getSqsUrl()).thenReturn(System.getProperty("tests.s3source.queue.url"));
+        when(sqsOptions.getVisibilityTimeout()).thenReturn(Duration.ofSeconds(60));
+        when(sqsOptions.getMaximumMessages()).thenReturn(10);
+        when(sqsOptions.getWaitTime()).thenReturn(Duration.ofSeconds(10));
+        when(s3SourceConfig.getSqsOptions()).thenReturn(sqsOptions);
+        when(s3SourceConfig.getOnErrorOption()).thenReturn(OnErrorOption.DELETE_MESSAGES);
+    }
+
+    private SqsWorker createObjectUnderTest() {
+        return new SqsWorker(sqsClient, s3Service, s3SourceConfig);
+    }
+
+    @AfterEach
```

Comment:
It may be better here to use `@AfterAll` which will run only after clearing all the items. 

Relatedly, we can do some setup in a `@BeforeAll` such as setting up the credentials.

This suite would probably also benefit from `PER_CLASS` instance lifecycle. https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-instance-lifecycle

Reply:
We should be using `@AfterEach` since each test is writing objects to S3 and we've to delete messages before running any other test.